contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -584,43 +626,64 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
     } else {
       gasWei = uint256(feedValue);
     }
-    (, feedValue, , timestamp, ) = i_linkNativeFeed.latestRoundData();
+    (, feedValue, , timestamp, ) = i_linkUSDFeed.latestRoundData();
+    if (
+      feedValue <= 0 || block.timestamp < timestamp || (staleFallback && stalenessSeconds < block.timestamp - timestamp)
+    ) {
+      linkUSD = s_fallbackLinkPrice;
+    } else {
+      linkUSD = uint256(feedValue);
+    }
+    (, feedValue, , timestamp, ) = i_nativeUSDFeed.latestRoundData();
```

Comment:
can this part use the _getNativeUSD function in L653 to reuse some code?

Reply:
good call!